pkg/test/environment.go

```diff
@@ -120,7 +120,7 @@ func NewEnvironment(ctx context.Context, env *coretest.Environment) *Environment
 	// Ensure we're able to hydrate the version before starting any reliant controllers.
 	// Version updates are hydrated asynchronously after this, in the event of a failure
 	// the previously resolved value will be used.
-	lo.Must0(versionProvider.UpdateVersion(ctx))
+	lo.Must0(versionProvider.UpdateVersionWithValidation(ctx))
```

Comment:
Do we need to do the version validation here now? Since we won't panic for the operator